### ISSUE TITLE
Training with multi-class and keypoints, the result is bad

### ISSUE THREAD

**Issue by yongguanjiangshan** [ultralytics/ultralytics#2674]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, thank you for the excellent job! But when i train the model with coco dataset(only use 6 class: person, bicycle, car, motorcycle, bus, truck) and only one keypoint for person(the nose). Which means the keypoint label is (0,0,0) for other class. When i train with below script, the result is bad. Could you give some idea to improve it? Looking forward to your reply! Thank you!!!
`
from ultralytics import YOLO

model = YOLO('yolov8n-pose_6cls_1point.yaml').load('yolov8n-pose.pt')  
model.train(data='coco-pose_6cls_1point.yaml', epochs=100, imgsz=640, device='0,1,2,3', batch=64)`

![image](https://github.com/ultralytics/ultralytics/assets/18051804/00d1a8cf-004e-41eb-8615-010384e6649e)
![image](https://github.com/ultralytics/ultralytics/assets/18051804/e1085529-9eab-4cf0-8eb4-f23b5302a989)

**Comment by glenn-jocher** [comment]
@yongguanjiangshan hello! Thank you for using YOLOv8. Based on the images you provided, it appears that the model is not detecting the objects correctly, even though you have trained it for 100 epochs. 

One possible reason for the poor performance may be the presence of only one keypoint for all classes. Another possible reason may be the use of fewer classes with respect to the original COCO dataset. To improve the performance, you may want to consider adding more keypoints, if possible, and increasing the number of classes. You could also try modifying the training hyperparameters such as the learning rate, batch size, and number of training epochs to see if they have an impact on the performance of the model.

I hope this helps! Let me know if you have any further questions.

**Comment by yongguanjiangshan** [comment]
> 
OK, thank you for your reply! I will try as your suggestions. Thanks again!

**Comment by glenn-jocher** [comment]
@yongguanjiangshan you're welcome! I hope the suggestions help to improve the performance of your YOLOv8 model. Don't hesitate to reach out if you have any further questions or concerns. Good luck with your training!
